src/credentials/providers/credentialsProviderId.ts

```diff
@@ -3,27 +3,28 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { CredentialSourceId } from '../../shared/telemetry/telemetry.gen'
+
 const CREDENTIALS_PROVIDER_ID_SEPARATOR = ':'
 
 export interface CredentialsProviderId {
-    /** Credential type name, e.g. "profile". */
-    readonly credentialType: string
+    /** Credential source id, e.g. "sharedCredentials". */
+    readonly credentialSource: CredentialSourceId
     /** User-defined profile name, e.g. "default". */
     readonly credentialTypeId: string
```

Comment:
Right now having this named 'credentialTypeId' is pretty confusing since it implies the ID is specific to the type of credentials, but it's actually specific to the source (e.g. it's a profile name when source is the shared credentials file). Perhaps dropping the 'Type' keyword would be beneficial.

Reply:
yeah, the naming in the entire subsystem is questionable. I also don't know why credentials are called "CredentialsProvider"...

Reply:
I think the naming is in reference to the SDK's Credential providers. But they have one provider for each of our `credentialType`. So I guess our providers are an abstraction that provides credentials from a `credentialSource`

Reply:
> So I guess our providers are an abstraction that provides credentials from a credentialSource, which is just a subset of valid credentialType

Our "providers" aren't really abstracting anything useful, it is only indirection. We also have "factories" and "managers".